app/api/suggestions/suggestions.ts

```diff
@@ -0,0 +1,164 @@
+              $cond: [
+                { $eq: ['$propertyName', 'title'] },
+                { v: [{ value: '$entity.title' }] },
+                {
+                  $arrayElemAt: [
+                    {
+                      $filter: {
+                        input: {
+                          $objectToArray: '$entity.metadata',
+                        },
+                        as: 'property',
+                        cond: {
+                          $eq: ['$$property.k', '$propertyName'],
+                        },
+                      },
+                    },
+                    0,
+                  ],
+                },
+              ],
+            },
+          },
+        },
+        {
+          $addFields: {
+            currentValue: { $arrayElemAt: ['$currentValue.v', 0] },
+          },
+        },
+        {
+          $addFields: {
+            currentValue: { $ifNull: ['$currentValue.value', ''] },
+          },
+        },
+        {
+          $addFields: {
+            state: {
+              $switch: {
+                branches: [
+                  {
+                    case: { $eq: ['$suggestedValue', '$currentValue'] },
+                    then: 'Matching',
+                  },
+                  {
+                    case: {
+                      $eq: ['$currentValue', ''],
+                    },
+                    then: 'Empty',
+                  },
+                ],
+                default: 'Pending',
+              },
+            },
+          },
+        },
+        ...(state ? [{ $match: { $expr: { $eq: ['$state', state] } } }] : []),
+        { $sort: { date: 1, state: -1 } },
+        {
+          $project: {
+            entityId: '$entity._id',
+            sharedId: '$entity.sharedId',
+            entityTitle: '$entity.title',
+            language: 1,
+            propertyName: 1,
+            suggestedValue: 1,
+            segment: 1,
+            currentValue: 1,
+            state: 1,
+            page: 1,
+          },
+        },
+      ],
+      {
+        stage1: [{ $group: { _id: null, count: { $sum: 1 } } }],
+        stage2: [{ $skip: offset }, { $limit: limit }],
+      },
+      { count: '$stage1.count', data: '$stage2' }
+    );
+
+    const totalPages = Math.ceil(count[0] / limit);
+    return { suggestions: data, totalPages };
+  },
+
+  accept: async (
+    acceptedSuggestion: {
+      _id: ObjectIdSchema;
+      sharedId: string;
+      entityId: string;
+    },
+    allLanguages: boolean,
+    params: { user: {}; language: string }
+  ) => {
+    const suggestion = await IXSuggestionsModel.getById(acceptedSuggestion._id);
+    if (!suggestion) {
+      throw new Error('Suggestion not found');
+    }
+
+    const query = allLanguages
+      ? { sharedId: acceptedSuggestion.sharedId }
+      : { sharedId: acceptedSuggestion.sharedId, _id: acceptedSuggestion.entityId };
+    const storedEntities = await entities.get(query, '+permissions');
```

Comment:
This is more of a question: we cannot use entities.multipleUpdate here, because that one updates the current language of multiple sharedId's, where here we want to go along "other axis", i.e. update one or more languages for the same sharedId?

Reply:
Yes, that's part of the reason. The main reason however is that the way multipleUpdate/entity.save work, they were overwriting the changes across the different languages. Additionally, it does not merit going into the details of the saving because we don't need here all the complexity that entity save has.

Reply:
Got it.

Reply:
Got it.